wrangler-service/src/main/java/co/cask/wrangler/service/explorer/FilesystemExplorer.java

```diff
@@ -210,17 +201,25 @@ private void loadFile(HttpServiceResponder responder, String scope, String path,
       properties.put(PropertyIds.SAMPLER_TYPE, SamplingMethod.NONE.getMethod());
       Format format = type == DataType.BINARY ? Format.BLOB : Format.TEXT;
       properties.put(PropertyIds.FORMAT, format.name());
-      ws.writeProperties(id, properties);
+      WorkspaceMeta workspaceMeta = WorkspaceMeta.builder(id, name)
+        .setScope(scope)
+        .setProperties(properties)
+        .build();
+      ws.writeWorkspaceMeta(workspaceMeta);
+
+      stream = new BufferedInputStream(location.getInputStream());
+      byte[] bytes = new byte[(int) location.length() + 1];
+      stream.read(bytes);
```

Comment:
do we always update workspace or check if data was read into buffer?

Reply:
I haven't changed the logic here, it never checked.